src/toxcore/io_tokio.rs

```diff
@@ -32,3 +32,51 @@ pub fn send_all_to<T: Send + 'static, S, Tx, E: Debug>(tx: &Tx, s: S) -> impl Fu
         .map(|_tx| ()) // ignore tx because it was cloned
 }
 
+/// `Sink` type that can either be empty or store inner `Sink`. When it's empty
+/// it drops all sent elements.
+#[derive(Debug, Clone)]
+pub struct OptionalSink<S: Sink>(Option<S>);
+
+impl<S: Sink> OptionalSink<S> {
```

Comment:
This PR looks good except I don't like OptionalSink in io_tokio.rs file. Move it somewhere

Reply:
OptionalSink and send_to/send_all_to have common topic - they both are auxiliary utils for futures. Also we are getting rid of `IoFuture`/`IoStream` in favor of `impl Future` and async/await in the future. So probably we can remain it together and just rename `io_rokio` mod somehow?

Reply:
Fine, let think about it later